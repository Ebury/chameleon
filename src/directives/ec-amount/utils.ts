
import defaultOptions from './options';
import type { AmountDirectiveOptions } from './types';

function format(input: undefined, opts?: AmountDirectiveOptions): undefined;
function format(input: null, opts?: AmountDirectiveOptions): null;
function format(input: string | number, opts?: AmountDirectiveOptions): string;
function format(input: string | number | undefined | null, opts: AmountDirectiveOptions = defaultOptions): string | null | undefined {
  /* eslint-disable no-param-reassign */
  if (input === '' || input === null || input === undefined) {
    return input;
  }

  if (typeof input === 'number') {
    input = input.toString();
  }

  const negative = input.startsWith('-') ? '-' : '';

  input = sanitizeInput(input, opts.decimalSeparator);
  if (opts.precision > 0) {
    input = keepOnlyOneDecimalSeparator(input, opts.decimalSeparator);
  } else {
    input = removeDecimal(input, opts.decimalSeparator);
  }
  const parts = input.split(opts.decimalSeparator);
  let integralPart = parseToStr(parts[0]);
  let fractionalPart = parts[1];

  if (fractionalPart && fractionalPart.length > opts.precision) {
    fractionalPart = fractionalPart.substring(0, opts.precision);
  }

  integralPart = addGroupingSeparators(integralPart, opts.groupingSeparator);
  const result = negative + joinIntegralAndFractionalParts(integralPart, fractionalPart, opts.decimalSeparator);
  return result;
}

function parseToStr(number: string): string {
  const parsed = Number.parseFloat(number);
  if (Number.isNaN(parsed)) {
    return '';
  }
  return parsed.toString();
}

function keepOnlyOneDecimalSeparator(number: string, decimalSeparator: string): string {
  if (number.includes(decimalSeparator)) {
    const split = number.split(decimalSeparator);
    return split[0] + decimalSeparator + split.slice(1).join('');
  }
  return number;
}

function removeDecimal(number: string, decimalSeparator: string): string {
  const split = number.split(decimalSeparator);
  return split[0];
}

function unFormat(input: string, groupingSeparator: string, decimalSeparator: string): string {
  if (!input) {
    return input;
  }

  return input
    .replace(new RegExp(`\\${groupingSeparator}`, 'g'), '')
    .replace(new RegExp(`\\${decimalSeparator}`, 'g'), '.');
}

function sanitizeInput(input: string, decimalSeparator: string): string {
  return input.replace(new RegExp(`[^0-9${decimalSeparator}]`, 'g'), '') || '';
}

function addGroupingSeparators(integralPart: string, separator: string): string {
  return integralPart.replace(/(\d)(?=(?:\d{3})+\b)/gm, `$1${separator}`);
}

function joinIntegralAndFractionalParts(integralPart: string, fractionalPart: string, decimalSeparator: string): string {
  return fractionalPart !== undefined ? integralPart + decimalSeparator + fractionalPart : integralPart;
}

/* c8 ignore start */
function setCursor(el: HTMLInputElement, position: number) {
  const setSelectionRange = function setSelectionRange() {
    el.setSelectionRange(position, position);
  };

  if (el === document.activeElement) {
    setSelectionRange();
    setTimeout(setSelectionRange, 1); // Android Fix
  }
}
/* c8 ignore stop */

export {
  format,
  sanitizeInput,
  setCursor,
  unFormat,
};
