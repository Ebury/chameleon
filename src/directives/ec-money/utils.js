import defaults from './options';

function format(input, opt = defaults) {
  if (input === '' || input === null || input === undefined) {
    return input;
  }

  if (typeof input === 'number') {
    input = input.toString();
  }

  const negative = input.includes('-') ? '-' : '';

  input = sanitizeInput(input, opt.decimalSeparator);
  input = keepOnlyOneDecimalSeparator(input, opt.decimalSeparator);
  const parts = input.split(opt.decimalSeparator);
  let integer = parseToStr(parts[0]);
  let decimal = parts[1];

  if (decimal && decimal.length > opt.precision) {
    decimal = decimal.substr(0, opt.precision);
  }

  integer = addGroupingSeparators(integer, opt.groupingSeparator);
  const result = negative + joinIntegerAndDecimal(integer, decimal, opt.decimalSeparator);
  return result;
}

function parseToStr(number) {
  /* eslint no-restricted-globals: "off" */
  const parsed = parseFloat(number);
  if (isNaN(parsed)) {
    return '';
  }
  return parsed.toString();
}

function keepOnlyOneDecimalSeparator(number, decimalSeparator) {
  if (number.includes(decimalSeparator)) {
    const split = number.split(decimalSeparator);
    return split[0] + decimalSeparator + split.slice(1).join('');
  }
  return number;
}

function unFormat(input, groupingSeparator, decimalSeparator) {
  /* eslint no-param-reassign: "off" */
  input = (`${input}`).replace(new RegExp(`\\${groupingSeparator}`, 'g'), '');
  input = (`${input}`).replace(new RegExp(`\\${decimalSeparator}`, 'g'), '.');
  return input;
}

function sanitizeInput(input, decimalSeparator) {
  return input.replace(new RegExp(`[^0-9${decimalSeparator}]`, 'g'), '') || '';
}

function addGroupingSeparators(integer, separator) {
  return integer.replace(/(\d)(?=(?:\d{3})+\b)/gm, `$1${separator}`);
}

function joinIntegerAndDecimal(integer, decimal, decimalSeparator) {
  return decimal !== undefined ? integer + decimalSeparator + decimal : integer;
}

function setCursor(el, position) {
  const setSelectionRange = function setSelectionRange() {
    el.setSelectionRange(position, position);
  };

  if (el === document.activeElement) {
    setSelectionRange();
    setTimeout(setSelectionRange, 1); // Android Fix
  }
}

// https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Creating_and_triggering_events#The_old-fashioned_way
function event(name) {
  const evt = document.createEvent('Event');
  evt.initEvent(name, true, true);
  return evt;
}

export {
  format,
  unFormat,
  setCursor,
  event,
  sanitizeInput,
};
