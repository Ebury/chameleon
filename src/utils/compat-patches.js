/* istanbul ignore file */

import {
  Dropdown as FvDropdown,
  TooltipDirective as FvTooltipDirective,
  VClosePopper,
  VTooltip,
} from 'floating-vue';

// we need to tell the @vue/compat that all components coming from floating-vue are Vue 3 components so it skips
// them when patching component instances. These components can't run in default mode for Vue 2, there is lots of
// errors coming from the console in that mode.
export default function applyCompatPatches() {
  const compatConfig = {
    MODE: 3,
    RENDER_FUNCTION: false,
  };

  VTooltip.compatConfig = compatConfig;
  VClosePopper.compatConfig = compatConfig;
  FvTooltipDirective.compatConfig = compatConfig;
  for (const component of Object.values(FvTooltipDirective.components)) {
    component.compatConfig = compatConfig;
  }

  FvDropdown.compatConfig = compatConfig;

  for (const component of Object.values(FvDropdown.components)) {
    component.compatConfig = compatConfig;
  }
}
