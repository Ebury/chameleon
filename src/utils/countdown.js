import mitt from 'mitt';

export default class Countdown {
  constructor() {
    const m = mitt();
    Object.assign(this, m);
  }

  reduceMinutesLeft(totalSeconds) {
    this.getminutes(totalSeconds);
    this.getseconds(totalSeconds, this.minutesLeft);
  }

  start(secondsToGo, showMinutes) {
    this.seconds = secondsToGo;
    this.secondsLeft = secondsToGo;
    this.currentTime = Math.ceil(Date.now() / 1000);
    this.startTime = Math.ceil(Date.now() / 1000);
    this.interval = setInterval(() => this.reduceSecondsLeft(showMinutes), 1000);
  }

  get timeDifference() {
    return this.currentTime - this.startTime;
  }

  reduceSecondsLeft(showMinutes) {
    this.currentTime = Math.ceil(Date.now() / 1000);
    this.totalSeconds = this.seconds - this.timeDifference;
    this.minutesLeft = null;
    this.secondsLeft = this.totalSeconds;

    if (showMinutes) {
      this.reduceMinutesLeft(this.totalSeconds);
    }

    this.emit('time-updated', {
      seconds: this.secondsLeft,
      minutes: this.minutesLeft,
      totalSeconds: this.totalSeconds,
    });

    if (this.totalSeconds <= 0) {
      this.stop();
      this.emit('time-expired');
    }
  }

  stop() {
    clearInterval(this.interval);
  }

  getminutes(totalSeconds) {
    this.minutesLeft = Math.floor(totalSeconds / 60);
  }

  getseconds(totalSeconds, minutes) {
    // take mins remaining (as seconds) away from total seconds remaining
    this.secondsLeft = totalSeconds - Math.round(minutes * 60);
  }
}
