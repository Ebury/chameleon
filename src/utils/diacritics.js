export function removeDiacritics(str) {
  if (typeof str === 'string' && str.normalize) {
    // https://stackoverflow.com/questions/990904/remove-accents-diacritics-in-a-string-in-javascript
    return str.normalize('NFD').replace(/[\u0300-\u036f]/g, '');
  }

  // IE11 users will not enjoy this feature:
  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize
  return str;
}
