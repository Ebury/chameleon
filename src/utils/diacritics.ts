export function removeDiacritics(str: string): string {
  if (str && str.normalize) {
    // https://stackoverflow.com/questions/990904/remove-accents-diacritics-in-a-string-in-javascript
    return str.normalize('NFD').replace(/[\u0300-\u036f]/g, '');
  }

  // IE11 users will not enjoy this feature:
  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize
  return str;
}
