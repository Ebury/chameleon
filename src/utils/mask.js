export function mask(text, maskSymbol = '*', visibleChars = 2) {
  if (!text) {
    throw new Error('Text is required');
  }
  if (!maskSymbol) {
    throw new Error('Mask symbol cannot be empty');
  }
  if (!visibleChars || visibleChars < 0) {
    throw new Error('Visible characters must be a number greater than zero');
  }

  const digitsToBeMasked = text.length - visibleChars;
  return maskSymbol.repeat(digitsToBeMasked) + text.substr(digitsToBeMasked);
}
