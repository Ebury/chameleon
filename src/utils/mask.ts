export function mask(text: string, maskSymbol: string = '*', visibleChars: number = 2): string {
  if (!text) {
    throw new Error('Text is required');
  }
  if (!maskSymbol) {
    throw new Error('Mask symbol cannot be empty');
  }
  if (!visibleChars || visibleChars < 0) {
    throw new Error('Visible characters must be a number greater than zero');
  }

  const digitsToBeMasked = text.length - visibleChars;
  return maskSymbol.repeat(digitsToBeMasked) + text.substring(digitsToBeMasked);
}
