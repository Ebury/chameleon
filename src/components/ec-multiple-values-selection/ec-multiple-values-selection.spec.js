import { mount, createLocalVue } from '@vue/test-utils';
import EcMultipleValuesSelection from './ec-multiple-values-selection.vue';
import { withMockedConsole } from '../../../tests/utils/console';

const selectAllText = 'Select all';
const items = [{
  value: 'Success',
  icon: {
    name: 'rounded-check',
    type: 'success',
  },
  text: 'Success',
}, {
  value: 'Partially paid',
  icon: {
    name: 'rounded-partial',
    type: 'success',
  },
  text: 'Name two',
}, {
  value: 'Cancelled',
  icon: {
    name: 'rounded-cancelled',
    type: 'error',
  },
  text: 'Cancelled',
}];

function mountEcMultipleValuesSelection(props, mountOpts) {
  return mount(EcMultipleValuesSelection, {
    propsData: {
      selectAllText,
      ...props,
    },
    ...mountOpts,
  });
}

function mountEcMultipleValuesSelectionAsTemplate(template, props, wrapperComponentOpts, mountOpts) {
  const localVue = createLocalVue();

  const Component = localVue.extend({
    components: { EcMultipleValuesSelection },
    template,
    ...wrapperComponentOpts,
  });

  return mount(Component, {
    localVue,
    propsData: { ...props },
    ...mountOpts,
  });
}
describe('EcMultipleValuesSelection', () => {
  it('should render correctly if all the required pros are passed', () => {
    const wrapper = mountEcMultipleValuesSelection({ items });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should throw an error if required prop label is missing', () => {
    withMockedConsole((errorSpy) => {
      mountEcMultipleValuesSelection();
      expect(errorSpy).toHaveBeenCalledTimes(1);
      expect(errorSpy.mock.calls[0][0]).toContain('Missing required prop: "items"');
    });
  });

  it(':selectedItems should return the correct value from prop "value"', () => {
    const testValue = [{ value: 'test value', name: 'test name' }];
    const wrapper = mountEcMultipleValuesSelection({ items, value: testValue });

    expect(wrapper.vm.selectedItems).toEqual(testValue);
  });

  it('should not be visibile if isSearchable is set to false', () => {
    const wrapper = mountEcMultipleValuesSelection({ items, isSearchable: false });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should emit selected items', async () => {
    const wrapper = mountEcMultipleValuesSelection({ items });
    await wrapper.findByDataTest('ec-multiple-values-selection__checkbox-select').findByDataTest('ec-checkbox__input').trigger('click');
    expect(wrapper.emitted('change')).toBeTruthy();
  });

  it('should deselected items', async () => {
    const wrapper = mountEcMultipleValuesSelectionAsTemplate(
      '<ec-multiple-values-selection v-model="selectedItems" :items="items" :is-searchable="false" :select-all-text="selectAllText" />',
      {},
      {
        data() {
          return {
            items,
            selectAllText,
            selectedItems: [{ value: 'Cancel', name: 'Cancel' }],
          };
        },
      },
    );
    await wrapper.findByDataTest('ec-multiple-values-selection__checkbox-deselect').findByDataTest('ec-checkbox__input').trigger('click');
    expect(wrapper.vm.selectedItems).toEqual([]);
  });

  it('should toggle (select/unselect) all options clicked', async () => {
    const wrapper = mountEcMultipleValuesSelectionAsTemplate(
      '<ec-multiple-values-selection v-model="selectedItems" :items="items" :select-all-text="selectAllText" :canSelectAll="true" />',
      {},
      {
        data() {
          return {
            items,
            selectedItems: [],
            selectAllText,
          };
        },
      },
    );

    await wrapper.findByDataTest('ec-multiple-values-selection__select-all').findByDataTest('ec-checkbox__input').trigger('click');
    expect(wrapper.element).toMatchSnapshot();
    await wrapper.findByDataTest('ec-multiple-values-selection__select-all').findByDataTest('ec-checkbox__input').trigger('click');
    expect(wrapper.element).toMatchSnapshot();
  });
});
