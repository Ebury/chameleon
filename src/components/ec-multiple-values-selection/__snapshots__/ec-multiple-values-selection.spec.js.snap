// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EcMultipleValuesSelection should not be visibile if isSearchable is set to false 1`] = `
<div
  class="ec-multiple-values-selection"
  data-test="ec-multiple-values-selection"
>
  <!---->
   
  <div
    class="ec-loading"
    data-test="ec-loading"
  >
    <!---->
     
    <div
      class="ec-loading__content ec-loading__content--is-transparent"
      data-test="ec-loading__content"
    >
      <div>
        <!---->
          
        <li
          class="ec-multiple-values-selection__value-wrapper"
          data-test="ec-multiple-values-selection__value-wrapper"
        >
          <div
            class="ec-checkbox ec-multiple-values-selection__checkbox"
            data-test="ec-multiple-values-selection__checkbox-select ec-checkbox"
          >
            <input
              class="ec-checkbox__input"
              data-test="ec-checkbox__input"
              id="ec-checkbox-26"
              type="checkbox"
            />
             
            <div
              class="ec-checkbox__label-checkbox-wrapper"
            >
              <span
                class="ec-checkbox__check-icon-wrapper"
              >
                <!---->
              </span>
               
              <label
                class="ec-checkbox__label"
                data-test="ec-checkbox__label"
                for="ec-checkbox-26"
              >
                <div
                  class="ec-multiple-values-selection__label-wrapper"
                >
                  <svg
                    class="ec-multiple-values-selection__icon ec-icon ec-icon--success"
                    height="24"
                    width="24"
                  >
                    <use
                      xlink:href="#ec-rounded-check"
                    />
                  </svg>
                   
                  <span
                    class="ec-multiple-values-selection__label-text"
                  >
                    Success
                  </span>
                </div>
              </label>
            </div>
             
            <!---->
          </div>
        </li>
        <li
          class="ec-multiple-values-selection__value-wrapper"
          data-test="ec-multiple-values-selection__value-wrapper"
        >
          <div
            class="ec-checkbox ec-multiple-values-selection__checkbox"
            data-test="ec-multiple-values-selection__checkbox-select ec-checkbox"
          >
            <input
              class="ec-checkbox__input"
              data-test="ec-checkbox__input"
              id="ec-checkbox-28"
              type="checkbox"
            />
             
            <div
              class="ec-checkbox__label-checkbox-wrapper"
            >
              <span
                class="ec-checkbox__check-icon-wrapper"
              >
                <!---->
              </span>
               
              <label
                class="ec-checkbox__label"
                data-test="ec-checkbox__label"
                for="ec-checkbox-28"
              >
                <div
                  class="ec-multiple-values-selection__label-wrapper"
                >
                  <svg
                    class="ec-multiple-values-selection__icon ec-icon ec-icon--success"
                    height="24"
                    width="24"
                  >
                    <use
                      xlink:href="#ec-rounded-partial"
                    />
                  </svg>
                   
                  <span
                    class="ec-multiple-values-selection__label-text"
                  >
                    Name two
                  </span>
                </div>
              </label>
            </div>
             
            <!---->
          </div>
        </li>
        <li
          class="ec-multiple-values-selection__value-wrapper"
          data-test="ec-multiple-values-selection__value-wrapper"
        >
          <div
            class="ec-checkbox ec-multiple-values-selection__checkbox"
            data-test="ec-multiple-values-selection__checkbox-select ec-checkbox"
          >
            <input
              class="ec-checkbox__input"
              data-test="ec-checkbox__input"
              id="ec-checkbox-30"
              type="checkbox"
            />
             
            <div
              class="ec-checkbox__label-checkbox-wrapper"
            >
              <span
                class="ec-checkbox__check-icon-wrapper"
              >
                <!---->
              </span>
               
              <label
                class="ec-checkbox__label"
                data-test="ec-checkbox__label"
                for="ec-checkbox-30"
              >
                <div
                  class="ec-multiple-values-selection__label-wrapper"
                >
                  <svg
                    class="ec-multiple-values-selection__icon ec-icon ec-icon--error"
                    height="24"
                    width="24"
                  >
                    <use
                      xlink:href="#ec-rounded-cancelled"
                    />
                  </svg>
                   
                  <span
                    class="ec-multiple-values-selection__label-text"
                  >
                    Cancelled
                  </span>
                </div>
              </label>
            </div>
             
            <!---->
          </div>
        </li>
      </div>
    </div>
  </div>
</div>
`;

exports[`EcMultipleValuesSelection should render correctly if all the required props are passed 1`] = `
<div
  class="ec-multiple-values-selection"
  data-test="ec-multiple-values-selection"
>
  <!---->
   
  <div
    class="ec-loading"
    data-test="ec-loading"
  >
    <!---->
     
    <div
      class="ec-loading__content ec-loading__content--is-transparent"
      data-test="ec-loading__content"
    >
      <div>
        <!---->
          
        <li
          class="ec-multiple-values-selection__value-wrapper"
          data-test="ec-multiple-values-selection__value-wrapper"
        >
          <div
            class="ec-checkbox ec-multiple-values-selection__checkbox"
            data-test="ec-multiple-values-selection__checkbox-select ec-checkbox"
          >
            <input
              class="ec-checkbox__input"
              data-test="ec-checkbox__input"
              id="ec-checkbox-3"
              type="checkbox"
            />
             
            <div
              class="ec-checkbox__label-checkbox-wrapper"
            >
              <span
                class="ec-checkbox__check-icon-wrapper"
              >
                <!---->
              </span>
               
              <label
                class="ec-checkbox__label"
                data-test="ec-checkbox__label"
                for="ec-checkbox-3"
              >
                <div
                  class="ec-multiple-values-selection__label-wrapper"
                >
                  <svg
                    class="ec-multiple-values-selection__icon ec-icon ec-icon--success"
                    height="24"
                    width="24"
                  >
                    <use
                      xlink:href="#ec-rounded-check"
                    />
                  </svg>
                   
                  <span
                    class="ec-multiple-values-selection__label-text"
                  >
                    Success
                  </span>
                </div>
              </label>
            </div>
             
            <!---->
          </div>
        </li>
        <li
          class="ec-multiple-values-selection__value-wrapper"
          data-test="ec-multiple-values-selection__value-wrapper"
        >
          <div
            class="ec-checkbox ec-multiple-values-selection__checkbox"
            data-test="ec-multiple-values-selection__checkbox-select ec-checkbox"
          >
            <input
              class="ec-checkbox__input"
              data-test="ec-checkbox__input"
              id="ec-checkbox-5"
              type="checkbox"
            />
             
            <div
              class="ec-checkbox__label-checkbox-wrapper"
            >
              <span
                class="ec-checkbox__check-icon-wrapper"
              >
                <!---->
              </span>
               
              <label
                class="ec-checkbox__label"
                data-test="ec-checkbox__label"
                for="ec-checkbox-5"
              >
                <div
                  class="ec-multiple-values-selection__label-wrapper"
                >
                  <svg
                    class="ec-multiple-values-selection__icon ec-icon ec-icon--success"
                    height="24"
                    width="24"
                  >
                    <use
                      xlink:href="#ec-rounded-partial"
                    />
                  </svg>
                   
                  <span
                    class="ec-multiple-values-selection__label-text"
                  >
                    Name two
                  </span>
                </div>
              </label>
            </div>
             
            <!---->
          </div>
        </li>
        <li
          class="ec-multiple-values-selection__value-wrapper"
          data-test="ec-multiple-values-selection__value-wrapper"
        >
          <div
            class="ec-checkbox ec-multiple-values-selection__checkbox"
            data-test="ec-multiple-values-selection__checkbox-select ec-checkbox"
          >
            <input
              class="ec-checkbox__input"
              data-test="ec-checkbox__input"
              id="ec-checkbox-7"
              type="checkbox"
            />
             
            <div
              class="ec-checkbox__label-checkbox-wrapper"
            >
              <span
                class="ec-checkbox__check-icon-wrapper"
              >
                <!---->
              </span>
               
              <label
                class="ec-checkbox__label"
                data-test="ec-checkbox__label"
                for="ec-checkbox-7"
              >
                <div
                  class="ec-multiple-values-selection__label-wrapper"
                >
                  <svg
                    class="ec-multiple-values-selection__icon ec-icon ec-icon--error"
                    height="24"
                    width="24"
                  >
                    <use
                      xlink:href="#ec-rounded-cancelled"
                    />
                  </svg>
                   
                  <span
                    class="ec-multiple-values-selection__label-text"
                  >
                    Cancelled
                  </span>
                </div>
              </label>
            </div>
             
            <!---->
          </div>
        </li>
      </div>
    </div>
  </div>
</div>
`;

exports[`EcMultipleValuesSelection should toggle (select/unselect) all options clicked 1`] = `
<div
  class="ec-multiple-values-selection"
  data-test="ec-multiple-values-selection"
>
  <!---->
   
  <div
    class="ec-loading"
    data-test="ec-loading"
  >
    <!---->
     
    <div
      class="ec-loading__content ec-loading__content--is-transparent"
      data-test="ec-loading__content"
    >
      <div>
        <div
          class="ec-multiple-values-selection__select-all"
        >
          <div
            class="ec-checkbox ec-multiple-values-selection__select-all--checkbox"
            data-test="ec-multiple-values-selection__select-all ec-checkbox"
          >
            <input
              class="ec-checkbox__input"
              data-test="ec-checkbox__input"
              id="ec-checkbox-57"
              type="checkbox"
            />
             
            <div
              class="ec-checkbox__label-checkbox-wrapper"
            >
              <span
                class="ec-checkbox__check-icon-wrapper ec-checkbox__check-icon-wrapper--checked"
              >
                <svg
                  class="ec-checkbox__check-icon ec-icon"
                  height="16"
                  width="16"
                >
                  <use
                    xlink:href="#ec-simple-check"
                  />
                </svg>
              </span>
               
              <label
                class="ec-checkbox__label"
                data-test="ec-checkbox__label"
                for="ec-checkbox-57"
              >
                Select all
              </label>
            </div>
             
            <!---->
          </div>
        </div>
         
        <li
          class="ec-multiple-values-selection__value-wrapper"
          data-test="ec-multiple-values-selection__checkbox-deselect"
        >
          <div
            class="ec-checkbox ec-multiple-values-selection__checkbox"
            data-test="ec-checkbox"
          >
            <input
              class="ec-checkbox__input"
              data-test="ec-checkbox__input"
              id="ec-checkbox-58"
              type="checkbox"
            />
             
            <div
              class="ec-checkbox__label-checkbox-wrapper"
            >
              <span
                class="ec-checkbox__check-icon-wrapper ec-checkbox__check-icon-wrapper--checked"
              >
                <svg
                  class="ec-checkbox__check-icon ec-icon"
                  height="16"
                  width="16"
                >
                  <use
                    xlink:href="#ec-simple-check"
                  />
                </svg>
              </span>
               
              <label
                class="ec-checkbox__label"
                data-test="ec-checkbox__label"
                for="ec-checkbox-58"
              >
                <div
                  class="ec-multiple-values-selection__label-wrapper"
                >
                  <svg
                    class="ec-multiple-values-selection__icon ec-icon ec-icon--success"
                    height="24"
                    width="24"
                  >
                    <use
                      xlink:href="#ec-rounded-check"
                    />
                  </svg>
                   
                  <span
                    class="ec-multiple-values-selection__label-text"
                  >
                    Success
                  </span>
                </div>
              </label>
            </div>
             
            <!---->
          </div>
        </li>
        <li
          class="ec-multiple-values-selection__value-wrapper"
          data-test="ec-multiple-values-selection__checkbox-deselect"
        >
          <div
            class="ec-checkbox ec-multiple-values-selection__checkbox"
            data-test="ec-checkbox"
          >
            <input
              class="ec-checkbox__input"
              data-test="ec-checkbox__input"
              id="ec-checkbox-60"
              type="checkbox"
            />
             
            <div
              class="ec-checkbox__label-checkbox-wrapper"
            >
              <span
                class="ec-checkbox__check-icon-wrapper ec-checkbox__check-icon-wrapper--checked"
              >
                <svg
                  class="ec-checkbox__check-icon ec-icon"
                  height="16"
                  width="16"
                >
                  <use
                    xlink:href="#ec-simple-check"
                  />
                </svg>
              </span>
               
              <label
                class="ec-checkbox__label"
                data-test="ec-checkbox__label"
                for="ec-checkbox-60"
              >
                <div
                  class="ec-multiple-values-selection__label-wrapper"
                >
                  <svg
                    class="ec-multiple-values-selection__icon ec-icon ec-icon--success"
                    height="24"
                    width="24"
                  >
                    <use
                      xlink:href="#ec-rounded-partial"
                    />
                  </svg>
                   
                  <span
                    class="ec-multiple-values-selection__label-text"
                  >
                    Name two
                  </span>
                </div>
              </label>
            </div>
             
            <!---->
          </div>
        </li>
        <li
          class="ec-multiple-values-selection__value-wrapper"
          data-test="ec-multiple-values-selection__checkbox-deselect"
        >
          <div
            class="ec-checkbox ec-multiple-values-selection__checkbox"
            data-test="ec-checkbox"
          >
            <input
              class="ec-checkbox__input"
              data-test="ec-checkbox__input"
              id="ec-checkbox-62"
              type="checkbox"
            />
             
            <div
              class="ec-checkbox__label-checkbox-wrapper"
            >
              <span
                class="ec-checkbox__check-icon-wrapper ec-checkbox__check-icon-wrapper--checked"
              >
                <svg
                  class="ec-checkbox__check-icon ec-icon"
                  height="16"
                  width="16"
                >
                  <use
                    xlink:href="#ec-simple-check"
                  />
                </svg>
              </span>
               
              <label
                class="ec-checkbox__label"
                data-test="ec-checkbox__label"
                for="ec-checkbox-62"
              >
                <div
                  class="ec-multiple-values-selection__label-wrapper"
                >
                  <svg
                    class="ec-multiple-values-selection__icon ec-icon ec-icon--error"
                    height="24"
                    width="24"
                  >
                    <use
                      xlink:href="#ec-rounded-cancelled"
                    />
                  </svg>
                   
                  <span
                    class="ec-multiple-values-selection__label-text"
                  >
                    Cancelled
                  </span>
                </div>
              </label>
            </div>
             
            <!---->
          </div>
        </li>
         
      </div>
    </div>
  </div>
</div>
`;

exports[`EcMultipleValuesSelection should toggle (select/unselect) all options clicked 2`] = `
<div
  class="ec-multiple-values-selection"
  data-test="ec-multiple-values-selection"
>
  <!---->
   
  <div
    class="ec-loading"
    data-test="ec-loading"
  >
    <!---->
     
    <div
      class="ec-loading__content ec-loading__content--is-transparent"
      data-test="ec-loading__content"
    >
      <div>
        <div
          class="ec-multiple-values-selection__select-all"
        >
          <div
            class="ec-checkbox ec-multiple-values-selection__select-all--checkbox"
            data-test="ec-multiple-values-selection__select-all ec-checkbox"
          >
            <input
              class="ec-checkbox__input"
              data-test="ec-checkbox__input"
              id="ec-checkbox-57"
              type="checkbox"
            />
             
            <div
              class="ec-checkbox__label-checkbox-wrapper"
            >
              <span
                class="ec-checkbox__check-icon-wrapper"
              >
                <!---->
              </span>
               
              <label
                class="ec-checkbox__label"
                data-test="ec-checkbox__label"
                for="ec-checkbox-57"
              >
                Select all
              </label>
            </div>
             
            <!---->
          </div>
        </div>
          
        <li
          class="ec-multiple-values-selection__value-wrapper"
          data-test="ec-multiple-values-selection__value-wrapper"
        >
          <div
            class="ec-checkbox ec-multiple-values-selection__checkbox"
            data-test="ec-multiple-values-selection__checkbox-select ec-checkbox"
          >
            <input
              class="ec-checkbox__input"
              data-test="ec-checkbox__input"
              id="ec-checkbox-58"
              type="checkbox"
            />
             
            <div
              class="ec-checkbox__label-checkbox-wrapper"
            >
              <span
                class="ec-checkbox__check-icon-wrapper"
              >
                <!---->
              </span>
               
              <label
                class="ec-checkbox__label"
                data-test="ec-checkbox__label"
                for="ec-checkbox-58"
              >
                <div
                  class="ec-multiple-values-selection__label-wrapper"
                >
                  <svg
                    class="ec-multiple-values-selection__icon ec-icon ec-icon--success"
                    height="24"
                    width="24"
                  >
                    <use
                      xlink:href="#ec-rounded-check"
                    />
                  </svg>
                   
                  <span
                    class="ec-multiple-values-selection__label-text"
                  >
                    Success
                  </span>
                </div>
              </label>
            </div>
             
            <!---->
          </div>
        </li>
        <li
          class="ec-multiple-values-selection__value-wrapper"
          data-test="ec-multiple-values-selection__value-wrapper"
        >
          <div
            class="ec-checkbox ec-multiple-values-selection__checkbox"
            data-test="ec-multiple-values-selection__checkbox-select ec-checkbox"
          >
            <input
              class="ec-checkbox__input"
              data-test="ec-checkbox__input"
              id="ec-checkbox-60"
              type="checkbox"
            />
             
            <div
              class="ec-checkbox__label-checkbox-wrapper"
            >
              <span
                class="ec-checkbox__check-icon-wrapper"
              >
                <!---->
              </span>
               
              <label
                class="ec-checkbox__label"
                data-test="ec-checkbox__label"
                for="ec-checkbox-60"
              >
                <div
                  class="ec-multiple-values-selection__label-wrapper"
                >
                  <svg
                    class="ec-multiple-values-selection__icon ec-icon ec-icon--success"
                    height="24"
                    width="24"
                  >
                    <use
                      xlink:href="#ec-rounded-partial"
                    />
                  </svg>
                   
                  <span
                    class="ec-multiple-values-selection__label-text"
                  >
                    Name two
                  </span>
                </div>
              </label>
            </div>
             
            <!---->
          </div>
        </li>
        <li
          class="ec-multiple-values-selection__value-wrapper"
          data-test="ec-multiple-values-selection__value-wrapper"
        >
          <div
            class="ec-checkbox ec-multiple-values-selection__checkbox"
            data-test="ec-multiple-values-selection__checkbox-select ec-checkbox"
          >
            <input
              class="ec-checkbox__input"
              data-test="ec-checkbox__input"
              id="ec-checkbox-62"
              type="checkbox"
            />
             
            <div
              class="ec-checkbox__label-checkbox-wrapper"
            >
              <span
                class="ec-checkbox__check-icon-wrapper"
              >
                <!---->
              </span>
               
              <label
                class="ec-checkbox__label"
                data-test="ec-checkbox__label"
                for="ec-checkbox-62"
              >
                <div
                  class="ec-multiple-values-selection__label-wrapper"
                >
                  <svg
                    class="ec-multiple-values-selection__icon ec-icon ec-icon--error"
                    height="24"
                    width="24"
                  >
                    <use
                      xlink:href="#ec-rounded-cancelled"
                    />
                  </svg>
                   
                  <span
                    class="ec-multiple-values-selection__label-text"
                  >
                    Cancelled
                  </span>
                </div>
              </label>
            </div>
             
            <!---->
          </div>
        </li>
      </div>
    </div>
  </div>
</div>
`;
